runtime/compiler/il/symbol/J9Symbol.cpp

```diff
@@ -271,6 +271,44 @@ J9::Symbol::getRecognizedField()
       return TR::Symbol::UnknownField;
    }
 
+/**
+ * Return the owning class name of this recognized field.
+ * Return null if this symbol does not have a recognized field.
+ */
+const char *
+J9::Symbol::owningClassNameCharsForRecognizedField(int32_t & length)
+   {
+   length = 0;
+   TR::Symbol::RecognizedField recognizedField = getRecognizedField();
+   if (TR::Symbol::UnknownField == recognizedField)
+      {
+      return NULL;
+      }
+   int i;
+   F *knownField;
+   for (i = 0, knownField = &recognizedFieldName_c[i];
+         knownField->id != TR::Symbol::UnknownField;
```

Comment:
is it always safe to deref knownField?

Reply:
I believe it is safe as the last table entry struct `{TR::Symbol::UnknownField}`, always has the first `id` field. As soon as `TR::Symbol::UnknownField` is read from `id` field, loop terminates and no other field is read from this entry,

Reply:
The deref would be made obviously correct by using a referenece type rather than a pointer for `knownField`

Reply:
Using a reference type.